envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/MetadataNodeGroupTest.kt

```diff
@@ -42,52 +47,75 @@ class MetadataNodeGroupTest {
     @Test
     fun `should assign to group with listed dependencies`() {
         // given
-        val nodeGroup = MetadataNodeGroup(outgoingPermissions = true)
+        val nodeGroup = MetadataNodeGroup(createSnapshotProperties(outgoingPermissions = true))
         val node = node(serviceDependencies = setOf("a", "b", "c"), ads = false)
 
         // when
         val group = nodeGroup.hash(node)
 
         // then
         assertThat(group).isEqualTo(
-            ServicesGroup(proxySettings = ProxySettings().with(serviceDependencies = setOf("a", "b", "c")), ads = false)
+            ServicesGroup(
+                proxySettings = ProxySettings().with(
+                    serviceDependencies = setOf(
+                        ServiceDependency("a"),
+                        ServiceDependency("b"),
+                        ServiceDependency("c")
+                    )
+                ),
+                ads = false
+            )
         )
     }
 
     @Test
     fun `should assign to group with all dependencies on ads`() {
         // given
-        val nodeGroup = MetadataNodeGroup(allServicesDependenciesValue = "*", outgoingPermissions = true)
+        val nodeGroup = MetadataNodeGroup(createSnapshotProperties(outgoingPermissions = true))
         val node = node(serviceDependencies = setOf("*"), ads = true)
 
         // when
         val group = nodeGroup.hash(node)
 
         // then
         assertThat(group).isEqualTo(
-            AllServicesGroup(proxySettings = ProxySettings().with(serviceDependencies = setOf("*")), ads = true)
+            AllServicesGroup(
+                proxySettings = ProxySettings().with(serviceDependencies = setOf(
+                    ServiceDependency("*"))
+                ),
+                ads = true
+            )
         )
     }
 
     @Test
     fun `should assign to group with listed dependencies on ads`() {
         // given
-        val nodeGroup = MetadataNodeGroup(outgoingPermissions = true)
+        val nodeGroup = MetadataNodeGroup(createSnapshotProperties(outgoingPermissions = true))
         val node = node(serviceDependencies = setOf("a", "b", "c"), ads = true)
 
         // when
         val group = nodeGroup.hash(node)
 
         // then
         assertThat(group).isEqualTo(
-            ServicesGroup(proxySettings = ProxySettings().with(serviceDependencies = setOf("a", "b", "c")), ads = true)
+            ServicesGroup(
+                proxySettings = ProxySettings().with(
+                    serviceDependencies = setOf(
+                        ServiceDependency("a"),
```

Comment:
It seems like you need a shortcut method for creating the set of string-named service dependencies.
Before there was just `setOf("a", "b", "c")` and now there's a repetitive multiline pattern. How about a vararg function call like `serviceDependencies("a", "b", "c")`?